### ISSUE TITLE
DSL substitution issues

### ISSUE THREAD

**Issue by forgedhallpass** [projectdiscovery/nuclei#2601]
Decision needed on how to handle nested DSL expressions or variable evaluations.

Should it be `{{first_method({{second_method({{variable}})}})}}` or `{{first_method(second_method(variable))}}`

Consider corner cases:
1. 
```
variables:
    var1: "test value"

{{first_method(var1)}} # but what if the user wants the string "var1" instead of "test value"?
```

2.
```
{{md5("{{test}}") # the user wants the "{{test}}" string to be hashed. Escaping the braces would be needed
```

Extensive integration and unit testing is mandatory to prevent regression issues, because the functionality affects a high number of templates.

**Comment by Mzack9999** [comment]
The preferred syntax is: `{{first_method(second_method(variable))}}`

**Comment by forgedhallpass** [comment]
> The preferred syntax is: `{{first_method(second_method(variable))}}`

Please address all the other points as well.

**Comment by Mzack9999** [comment]
> > The preferred syntax is: `{{first_method(second_method(variable))}}`
> 
> We need to address all the other points as well.

Obviously, the previous comment was a starting point note left to expand the discussion further, as discussed internally.

**Comment by Mzack9999** [comment]
## Other corners cases
### Case 1
```yaml
variables:
    var1: "test value"

{{first_method(var1)}} # what if the user wants the string "var1" instead of "test value"?
```
Response:
```yaml
{{"var1"}}
```

**Comment by Mzack9999** [comment]
Depends on https://github.com/projectdiscovery/nuclei/pull/2679

**Comment by Ice3man543** [comment]
@Mzack9999 my PR solves the mentioned issue. Should we proceed with that implementation?

**Comment by dogancanbakir** [comment]
Closing this issue due to prolonged inactivity. Please reopen if you think this is still relevant or requires further discussion.
